File: backend/src/main/java/io/github/nicolasdesnoust/lambda/contact/MessageFormatter.java
package io.github.nicolasdesnoust.lambda.contact;

public class MessageFormatter {
    // The HTML body for the email.
    public String formatMessageAsHtml(ContactMessage message) {
        return "<h1>" + message.getSubject() + "</h1>"
                + "<p>" + message.getBody() + "</p>"
                + "<p>Ce message a été envoyé par " + message.getSender().getName() + " " + message.getSender().getEmailAddress() + "</p>"
                + "<small>This email was sent with <a href='https://aws.amazon.com/ses/'>"
                + "Amazon SES</a> using the <a href='https://aws.amazon.com/sdk-for-java/'>"
                + "AWS SDK for Java</a></small>";
    }

    // The email body for recipients with non-HTML email clients.
    public String formatMessageAsText(ContactMessage message) {
        return message.getSubject() + '\n' +
                message.getBody() + '\n' +
                "De : " + message.getSender().getName() + '\n' +
                message.getSender().getEmailAddress() + '\n';
    }
}

File: backend/functions/send-contact-mail/src/domain/message-formatter.ts
import { ContactMessage } from "./contact-message";

export class MessageFormatter {
  // The HTML body for the email.
  public formatMessageAsHtml(message: ContactMessage): string {
    return (
      "<h1>" +
      message.subject +
      "</h1>" +
      "<p>" +
      message.body +
      "</p>" +
      "<p>Ce message a été envoyé par " +
      message.sender.name +
      " " +
      message.sender.emailAddress +
      "</p>" +
      "<small>This email was sent with <a href='https://aws.amazon.com/ses/'>" +
      "Amazon SES</a> using the <a href='https://aws.amazon.com/fr/sdk-for-javascript/'>" +
      "AWS SDK for Typescript</a></small>"
    );
  }

  // The email body for recipients with non-HTML email clients.
  public formatMessageAsText(message: ContactMessage): string {
    return (
      message.subject +
      "\n" +
      message.body +
      "\n" +
      "De : " +
      message.sender.name +
      "\n" +
      message.sender.emailAddress +
      "\n"
    );
  }
}
